pkg/k8s/k8s_pod.go

```diff
@@ -314,6 +315,24 @@ func (c *Client) getPod(ctx context.Context, name string) (*v1.Pod, error) {
 	return pod, nil
 }
 
+func (c *Client) WaitForDeployment(ctx context.Context, name string) error {
+	for {
+		deployment, err := c.clientset.AppsV1().Deployments(c.namespace).Get(ctx, name, metav1.GetOptions{})
+		if err == nil && deployment.Status.ReadyReplicas > 0 {
+			break
+		}
+
+		select {
+		case <-ctx.Done():
+			return ErrWaitingForDeployment.WithParams(name).Wrap(err)
+		case <-time.After(waitRetry):
+			// Retry after some seconds
+		}
+	}
+
+	return nil
+}
```

Comment:
Can we move that to a new file called `k8s_deplotment.go`?

Reply:
sure it will be cleaner